realm/realm-library/src/androidTest/java/io/realm/DynamicRealmTests.java

```diff
@@ -664,4 +668,18 @@ public void run() throws Exception {
             }
         });
     }
+
+    @Test
+    public void equalTo_noFieldObjectShouldThrow() {
```

Comment:
This still tests a non-existing field on an existing Table, but the issue reported was about an (obvious) non-existing field on a non-existing table.